plenum/server/node.py

```diff
@@ -1613,9 +1615,12 @@ def is_catchup_needed(self) -> bool:
             if self.has_ordered_till_last_prepared_certificate():
                 logger.debug('{} ordered till last prepared certificate'.format(self))
                 return False
-            if self.catchup_rounds_without_txns >= self.config.MAX_CATCHUPS_DONE_DURING_VIEW_CHANGE:
-                logger.debug('{} has completed {} catchup rounds'.
-                             format(self, self.catchup_rounds_without_txns))
+
+            ts_since_catch_up_start = time.perf_counter() - self._catch_up_start_ts
```

Comment:
Abstract this logic around `catchup_rounds_without_txns` and `_catch_up_start_ts` in a method